packages/strapi-plugin-content-manager/admin/src/components/InputUID/RightLabel.js

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Success, Remove } from '@buffetjs/icons';
+import styled from 'styled-components';
+import { useGlobalContext } from 'strapi-helper-plugin';
+
+import pluginId from '../../pluginId';
+
+// This component is only used in this file so there is no need to create a separated file.
+const RightContentLabel = styled.div`
+  padding: 0 5px;
+  text-transform: capitalize;
+  color: ${props => props.theme.main.colors[props.color]};
+`;
+
+const RightLabel = ({ label, availability }) => {
+  const { formatMessage } = useGlobalContext();
+
+  if (label) {
+    return (
+      <RightContentLabel color="blue">
+        {formatMessage({
+          id: `${pluginId}.components.uid.regenerate`,
+        })}
+      </RightContentLabel>
+    );
+  }
+
+  if (availability !== null) {
+    // This should be more generic in the futur.
+    return availability.isAvailable ? (
```

Comment:
You can refactor your statement here:

```
const { isAvailable } = availability;
const message = formatMessage({ id: getTrad(`component.uid.${isAvailable ? "" : "un"}availbale`) });
```

Then, you can use the message in your `RightContentLabel`, as the icons components differs a lot between the `isAvailable` property I don't think that is relevant to refactor them.

Reply:
If I do that, the ckeck, will be done twice. the first time for icons and the second for the right label. Do you still want to do this?

Reply:
No that's fine leave it this way! In fact I was thinking that we could refactor the statement but then I saw the rendered components differs too much from one to another.